File: python/S15_GRAPH/L1_BASIC/1_graph.py
class graph:
    def __init__(self,V):
        self.matrix=[[0 for _ in range(V)] for _ in range(V)]
        self.V=V
    def addEdge(self,src,dst):
        self.matrix[src][dst]=1
        self.matrix[dst][src]=1
    def display(self):
        for i in range(self.V):
            for j in range(self.V):
                print(self.matrix[i][j],end=" ")
            print(end="\n")
    
if __name__=="__main__":
    g=graph(5)
    g.addEdge(0,1)
    g.addEdge(1,2)
    g.addEdge(2,3)
    g.addEdge(3,4)
    g.display()


            

File: JAVA/S15_GRAPH/L1_BASIC/_1_graph.java
class Node{
    int data;
    Node next;
    Node(int data){
        this.data=data;
        next=null;
    }
}
class graph{
    Node adjlist;
    int numsize;
    graph(numsize){
        this.numsize=numsize;
        for(int i=0;i<numsize;i++){
            adjlist[i]=null;
        }
    }
}
class graph2{

    public void addEdge(graph g,int src,int dst){
        Node newnode=new Node(dst);
        newnode.next=g.adjlist[src];
        g.adjlist[src]=newnode;
        newnode=new Node(src);
        newnode.next=g.adjlist[dst];
        g.adjlist[dst]=newnode;

    }
}
public class _1_graph{
    graph2 g=new graph2();
    graph g2=new graph(5);
    g.addEdge(g2,0,1);
    g.addEdge(g2,1,2);
    g.addEdge(g2,2,3);
    g.addEdge(g2,3,4);
}